src/core/ext/filters/client_channel/http_proxy_mapper.cc

```diff
@@ -72,23 +76,27 @@ bool ServerInCIDRRange(const grpc_resolved_address& server_address,
   uint32_t mask_bits = 0;
   if (absl::SimpleAtoi(possible_cidr.second, &mask_bits)) {
     grpc_sockaddr_mask_bits(&*proxy_address, mask_bits);
-    return grpc_sockaddr_match_subnet(&server_address, &*proxy_address,
+    return grpc_sockaddr_match_subnet(&*server_address, &*proxy_address,
                                       mask_bits);
   }
   return false;
 }
 
+bool ExactMatchOrSubdomain(absl::string_view host_name,
+                           absl::string_view host_name_or_domain) {
+  return absl::EndsWithIgnoreCase(
+      host_name, absl::StripAsciiWhitespace(host_name_or_domain));
+}
+
+// Parses the list of host names, addresses or subnet masks and returns true if
+// the target address or host matches any value.
 bool AddressIncluded(
     const absl::StatusOr<grpc_resolved_address>& target_address,
-    absl::string_view host_name, absl::string_view address_ranges) {
-  for (const auto& entry :
-       absl::StrSplit(address_ranges, ',', absl::SkipEmpty())) {
-    auto address_or_mask = absl::StripAsciiWhitespace(entry);
-    if (absl::EndsWithIgnoreCase(host_name, address_or_mask)) {
-      return true;
-    }
-    if (target_address.ok() &&
-        ServerInCIDRRange(*target_address, address_or_mask)) {
+    absl::string_view host_name, absl::string_view addresses_and_subnets) {
+  for (absl::string_view entry :
+       absl::StrSplit(addresses_and_subnets, ',', absl::SkipEmpty())) {
+    if (ExactMatchOrSubdomain(host_name, entry) ||
```

Comment:
I think we should call `absl::StripAsciiWhitespace()` just once here, rather than doing it separately in each of `ExactMatchOrSubdomain()` and `ServerInCIDRRange()`.